src/ConfigTool.cpp

```diff
@@ -27,84 +32,182 @@ void ConfigTool::load() {
 	}
-
 	result += continueHtml;
 
 	for (auto item : variables_) {
-		result += "<div class=\"form-group row\"><div class=\"col-2\"><label>" + item.first + "</label></div><div class=\"col-10\"><input name=\"" + item.first + "\" class=\"form-control\" type=\"text\" value=\"" + item.second->toString() + "\" /></div></div>";
+		
+		if (item.second->hideInWeb) continue;
+		
+		result += 
+			"<tr>"
+				"<td><label>" + item.first + "</label></td>"
+				"<td>";
+		
+		switch (item.second->varType()) {
+			
+			case VT_TEXT:
+				result += createInput(item.second, "text");
+				//<input name=\"" + item.first + "\" type=\"text\" value=\"";
+				//if (!item.second->hideValueInWeb) result += item.second->toString();
+				//result += "\" />";
+				break;
+			
+			case VT_NUM:
+				result += createInput(item.second, "number");
+				//result += "<input name=\"" + item.first + "\" type=\"number\" value=\"";
+				//if (!item.second->hideValueInWeb) result += item.second->toString();
+				//result += "\" />";
+				break;
+			
+			case VT_IP:
+				result += createInput(item.second, "text");
+				break;
+			
+			case VT_BOOL:
+				bool val = *(((ConfigVar<bool>*)item.second)->pointer);
+				result += "<fieldset>";
+				result += createBoolSelector((ConfigVar<bool>*)item.second, "true",  "on",  val  && !item.second->hideValueInWeb);
+				result += createBoolSelector((ConfigVar<bool>*)item.second, "false", "off", !val && !item.second->hideValueInWeb);
+//				result += createBoolTag(item.first, "true", "on", val && !item.second->hideValueInWeb);
+//				result += createBoolTag(item.first, "false", "off", !val && !item.second->hideValueInWeb);
+				result += "</fieldset>";
+				break;
+		
+		}
+				
+		result +=
+				"</td>"
+			"</tr>";
 	}
 
+/*
+	result+="<tr><td>";
+	File f = SPIFFS.open("/config.json", "r");
+	while (f and f.available()) {
+		result += f.read();
+	}
+	result+="</td></tr>";
+*/
+	
 	result += endHtml;
 
+	//Serial.println(result);
+
 	return result;
 }
 
-std::function<void()> ConfigTool::getWebHandler(ESP8266WebServer* server) {
+std::function<void()> ConfigTool::getWebHandler(WebServer* server) {
 	return [this, server]() {
 		bool updated = false;
 
-		if (server->args() == 1 && server->hasArg("reset") && server->arg("reset") == "true") {
+		if (/*server->args() == 1 &&*/ server->hasArg("reset")) { //} && server->arg("reset") == "true") {
 			Serial.println("Reset is true.");
 			for (auto item : variables_) {
 				item.second->reset();
 			}
 			updated = true;
 		}
 		else {
-			Serial.write("Args hit: ");
 			for (int i = 0; i < server->args(); i++) {
 				String name = server->argName(i);
+				
 				auto item = variables_.find(name);
-				Serial.print(name + " | ");
 				if (item == variables_.end()) {
 					continue;
 				}
+				String val = server->arg(name);
+				
+				Serial.println(name + "=" + val);
+				
+				// skip empty input for values that are not displayed
+				if (item->second->hideValueInWeb && val == "") {
```

Comment:
What happens when `item->second->hideValueInWeb` is `true` but the value is not `""`? Can that happen?